### PR TITLE
docs: Document how and when we update the assets

### PR DESCRIPTION
**Pull request by jcvenegas**
Describe how are updated:

linux-container for the kernel
clear-containers-image for the rootfs image

Fixes: #632

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by jcvenegas** [comment]
related to https://github.com/clearcontainers/packaging/issues/165#issuecomment-337853076

**Comment by jcvenegas** [comment]
Hi @jodh-intel sorry for the delay X( .  Could you take a look to this document and ask any gap there is in update process.

File: docs/assets-update.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+The Clear Containers image guest OS system is generated using Clear Linux packages. This image 
+Contain a minimal system for Clear Containers. The image is generated twice a day as
+part of the Clear Linux Release Cycle. 
+
+### Update Clear Containers image.
+
+The Clear Containers image should be updated only when critical updates were
+done to the packages used by Clear Containers Guess OS:
+
+- Systemd
```
Comment (jodh-intel) [inline docs/assets-update.md]: Should this be `systemd` or just "systemd". @iphutch will know ;)

File: docs/assets-update.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- Clear Containers Agent
```
Comment (jodh-intel) [inline docs/assets-update.md]: Can you make this "[Clear Containers Agent](https://github.com/clearcontainers/agent)".

File: docs/assets-update.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+in the [Clear Containers Release](https://github.com/clearcontainers/runtime/wiki/Release-Process).
+
+### Owner
+
+Carlos Venegas (@jcvenegas) current Clear Containers Release owner will verify
+this each release.
+
+## Clear Containers Continues integration.
+
+Official Clear Containers Packages are hosted in OBS build system. In order to
```
Comment (jodh-intel) [inline docs/assets-update.md]: s/in OBS/in the OBS/

File: docs/assets-update.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+avoid availability issues the kernel and image for Clear Containers Continues
```
Comment (jodh-intel) [inline docs/assets-update.md]: s/Continues/continuous/

File: docs/assets-update.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+### Clear Containers Image.
+
+The Clear Containers image guest OS system is generated using Clear Linux packages. This image 
+Contain a minimal system for Clear Containers. The image is generated twice a day as
+part of the Clear Linux Release Cycle. 
+
+### Update Clear Containers image.
+
+The Clear Containers image should be updated only when critical updates were
+done to the packages used by Clear Containers Guess OS:
```
Comment (jodh-intel) [inline docs/assets-update.md]: s/by Clear Containers Guess OS/by the Clear Containes guest OS/.

File: docs/assets-update.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+Official Clear Containers Packages are hosted in OBS build system. In order to
+avoid availability issues the kernel and image for Clear Containers Continues 
+integration system used is downloaded from Clear Linux\* packages:
+
+```
+https://download.clearlinux.org/releases/${CLEAR_VERSION}/clear/x86_64/os/Packages/linux-container-${KERNEL_VERSION}.rpm 
+```
+
+The packages from Clear Linux provide the same kernel and image used in other
+Linux Distributions. And can be  safely used without any compatibility issue.
```
Comment (jodh-intel) [inline docs/assets-update.md]: I'd change these two sentences to something like, "The Clear Containers assets packages provide identica kernel and images for the various Linux Distributions." I'm not sure we need the 2nd sentence tbh.

File: docs/assets-update.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+https://github.com/clearcontainers/packaging/blob/master/scripts/get-image-changes.sh
+
+The image must be updated in [Clear Containers packaging repository](https://github.com/clearcontainers/packaging)
+and defined in https://github.com/clearcontainers/runtime/blob/master/versions.txt
+in the [Clear Containers Release](https://github.com/clearcontainers/runtime/wiki/Release-Process).
+
+## Clear Containers Continuous integration.
+
+Official Clear Containers Packages are hosted in OBS build system. In order to
+avoid availability issues the kernel and image for Clear Containers continuous
```
Comment (iphutch) [inline docs/assets-update.md]: image for the Clear ...

File: docs/assets-update.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+Official Clear Containers Packages are hosted in OBS build system. In order to
+avoid availability issues the kernel and image for Clear Containers continuous 
+integration system used is downloaded from Clear Linux\* packages:
+
+```
+https://download.clearlinux.org/releases/${CLEAR_VERSION}/clear/x86_64/os/Packages/linux-container-${KERNEL_VERSION}.rpm 
+```
+
+The Clear Containers assets packages provide identica kernel and images
```
Comment (iphutch) [inline docs/assets-update.md]: typo identical

File: docs/assets-update.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+for the various Linux Distributions.
+
+### Security
+
+In case of any critical update is done in the kernel or Clear Contaienrs image
```
Comment (iphutch) [inline docs/assets-update.md]: In the case of critical updates to the kernel or the Clear Containers image, a new Clear Containers release will be rolled out immediately.

File: docs/assets-update.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+a new Clear Containers Release will need to be done immediately otherwise
+the next Clear Containers release will include the changes done in the week.
```
Comment (iphutch) [inline docs/assets-update.md]: I don't understand. The release will be immediate or in a weeks time?

File: docs/assets-update.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
+# Clear Containers assets updates 
+
```
[the remark this answers is not in the document]
Comment (jcvenegas) [reply docs/assets-update.md]: Done, I just mention in the image section about CL packages, the kernel is not produced from CL anymore.

**Comment by jcvenegas** [comment]
@iphutch thanks for the review.
`is this document about when things should happen, or how things happen now?`
It things happen now.  
`We go back and fourth between future and present tense in this doc`
I moved most of the document to use present tense

File: docs/assets-update.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+Clear Containers requires additional resources to create a virtual machine
+container. These resources are called  [Clear Container assets](https://github.com/clearcontainers/runtime/blob/master/docs/architecture/architecture.md#assets)
+which are a `kernel` and a root `filesystem` image. This document describe
```
Comment (iphutch) [inline docs/assets-update.md]: s/describe/describes

File: docs/assets-update.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+container. These resources are called  [Clear Container assets](https://github.com/clearcontainers/runtime/blob/master/docs/architecture/architecture.md#assets)
+which are a `kernel` and a root `filesystem` image. This document describe 
+when these components are updated.
+
+## Clear Containers Kernel
+
+The [Clear Containers kernel](https://github.com/clearcontainers/linux) is a
+Linux\* kernel based on the latest vanilla version of the [LTS kernel](https://www.kernel.org/) and 
+the includes [patches](https://github.com/clearlinux-pkgs/linux-container/)
+necessary to run Clear Containers. The `LTS` branch is more "stable" and tested than latest `stable` branch kernel.
```
Comment (iphutch) [inline docs/assets-update.md]: * that the latest
* line length (78chars)

File: docs/assets-update.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+Using this `LTS` version ensures fewer necessary updates.
+
+### Clear Containers Kernel update
+
+Each time a new kernel version is rolled out it is updated
+in the Clear Containers packaging repository and the [Clear Containers Linux kernel](https://github.com/clearcontainers/linux). 
+On each Clear Containers Release the latest version in this repository is used as recommended kernel for
```
Comment (iphutch) [inline docs/assets-update.md]: * release
* the recommended
* line length

File: docs/assets-update.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+Clear Containers requires additional resources to create a virtual machine
+container. These resources are called  [Clear Container assets](https://github.com/clearcontainers/runtime/blob/master/docs/architecture/architecture.md#assets)
+which are a `kernel` and a root `filesystem` image. This document describe 
+when these components are updated.
+
+## Clear Containers Kernel
+
+The [Clear Containers kernel](https://github.com/clearcontainers/linux) is a
+Linux\* kernel based on the latest vanilla version of the [LTS kernel](https://www.kernel.org/) and
```
Comment (iphutch) [inline docs/assets-update.md]: wrap "and" to the next line

File: docs/assets-update.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+The Clear Containers image is updated only when critical updates are
+done to the packages used by the Clear Containers guest OS:
+
+- Systemd
+- [Clear Containers Agent](https://github.com/clearcontainers/agent)
+- iptables
+- core-utils
+
+This is verified each release using the [`get-image-changes.sh`](https://github.com/clearcontainers/packaging/blob/master/scripts/get-image-changes.sh) script.
```
Comment (iphutch) [inline docs/assets-update.md]: wrap "script." to the next line

File: docs/assets-update.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+The image must be updated in the [Clear Containers packaging repository](https://github.com/clearcontainers/packaging)
+and defined in the [versions.txt](https://github.com/clearcontainers/runtime/blob/master/versions.txt) file.
+
+## Clear Containers Continuous integration
+
+Official Clear Containers Packages are hosted in OBS build system. In order to
```
Comment (iphutch) [inline docs/assets-update.md]: in the OBS build...

**Comment by jcvenegas** [comment]
@iphutch  changes applied and  wrapped text at 80 (when possible, missing parts with URLs)

File: docs/assets-update.md [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+container. These resources are called
+[Clear Container assets](https://github.com/clearcontainers/runtime/blob/master/docs/architecture/architecture.md#assets)
+which are a `kernel` and a root `filesystem` image. This document describes when
+these components are updated.
+
+## Clear Containers Kernel
+
+The [Clear Containers kernel](https://github.com/clearcontainers/linux) is a
+Linux\* kernel based on the latest vanilla version of the 
+[LTSkernel](https://www.kernel.org/) and the includes
```
Comment (iphutch) [inline docs/assets-update.md]: Missing a space: LTS kernel

File: docs/assets-update.md [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+[patches](https://github.com/clearlinux-pkgs/linux-container/) necessary to run
+Clear Containers. The `LTS` branch is more "stable" and tested that the latest
```
Comment (iphutch) [inline docs/assets-update.md]: Can you replace the word "stable" with something more descriptive?

Comment (jcvenegas) [reply docs/assets-update.md]: @iphutch  hi, updated
Let me know if this is more clean ( it is already in the  in the PR ) :

 The `LTS` branch is only updated important bugfixes and it is
 more tested that the latest `stable` branch kernel. Using this `LTS` version
 ensures fewer necessary updates.

Comment (iphutch) [reply docs/assets-update.md]: The `LTS` branch undergoes more testing than the latest `stable` branch of the kernel. The `LTS` branch is only updated with important bug fixes and in turn using this branch ensures fewer required updates.

**Comment by sameo** [comment]
LGTM

**Pull request merged**